src/core/server/services/users/user.updateRole.spec.ts

```diff
@@ -0,0 +1,209 @@
+  tenantID,
+});
+const orgMod = createUserFixture({
+  role: GQLUSER_ROLE.MODERATOR,
+  tenantID,
+});
+const site1Mod = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.MODERATOR,
+  moderationScopes: {
+    siteIDs: [site1.id],
+  },
+});
+const site1Staff = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.STAFF, // Hmm, the way we determine staff status in src/server/services/comment/pipeline/phases/tagStaff.ts seems to just tag site mods?
+  moderationScopes: {
+    siteIDs: [site1.id],
+  },
+});
+const site1Member = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.MEMBER,
+  membershipScopes: {
+    siteIDs: [site1.id],
+  },
+});
+const site1Commenter = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.COMMENTER,
+});
+const users = [
+  admin,
+  orgMod,
+  site1Mod,
+  site1Staff,
+  site1Member,
+  site1Commenter,
+];
+
+const inputStr = (
+  viewerRole: GQLUSER_ROLE,
+  userFromRole: GQLUSER_ROLE,
+  userToRole: GQLUSER_ROLE
+) =>
+  `viewer role = ${viewerRole}, user from role = ${userFromRole},  userToRole = ${userToRole}`;
+
+describe("updateRole", () => {
+  let mongo: MongoContext;
+
+  /* eslint-disable-next-line */
+  require("coral-server/models/user").retrieveUser.mockImplementation(
+    async (_mongo: any, _tenantID: any, userID: string) =>
+      users.find(({ id }) => id === userID)
+  );
+
+  /* eslint-disable-next-line */
+  require("coral-server/models/user").updateUserRole.mockImplementation(
+    async (
+      _mongo: any,
+      _tenantID: any,
+      userID: string,
+      role: GQLUSER_ROLE
+    ) => ({
+      ...users.find(({ id }) => id === userID),
+      role,
+    })
+  );
+
+  let uut: (
+    mongo: MongoContext,
+    tenant: Tenant,
+    viewer: User,
+    user: User,
+    role: GQLUSER_ROLE
+  ) => ReturnType<typeof updateRole>;
+
+  const exhaustiveInputs = new Map<string, boolean>();
+  beforeAll(() => {
+    ({ ctx: mongo } = createMockMongoContex());
+
+    // Wrapping our function to track what inputs have been tested
+    uut = (mongoCtx, tenant, viewer, user, role) => {
+      const inputHash = inputStr(viewer.role, user.role, role);
+      exhaustiveInputs.set(inputHash, true);
+
+      return updateRole(mongo, tenant, viewer, user.id, role);
+    };
+
+    for (const viewer of users) {
+      for (const user of users) {
+        for (const newRole of ROLES) {
+          const inputHash = inputStr(viewer.role, user.role, newRole);
+          exhaustiveInputs.set(inputHash, false);
+        }
+      }
+    }
+  });
+
+  it("allows admins to assign any user any role", async () => {
+    const nonAdmins = users.filter(({ role }) => role !== GQLUSER_ROLE.ADMIN);
+    for (const user of nonAdmins) {
+      for (const role of ROLES) {
+        const res = await uut(mongo, mockTenant, admin, user, role);
+
+        expect(res.role).toEqual(role);
+      }
+    }
+  });
+
+  it("TODO: is this ok?! allows organization moderators to allocate site mods", async () => {
```

Comment:
Did you want to update this todo? Looks like it's testing what we want it to, that org mods can add site mods?